OData/test/UnitTest/System.Web.OData.Test/OData/Formatter/NullCollectionsTests.cs

```diff
@@ -0,0 +1,341 @@
+            }
+
+            return TestObject;
+        }
+    }
+
+    public class NullCollectionsTestsModel
+    {
+        public NullCollectionsTestsModel()
+        {
+            Array = new int[] { 42 };
+            IEnumerable = new int[] { 42 };
+            ICollection = new int[] { 42 };
+            IList = new int[] { 42 };
+            List = new List<int> { 42 };
+            Collection = new Collection<int> { 42 };
+            ComplexCollection = new NullComplexChild[] { new NullComplexChild { A = 42 } };
+            ParentComplex = new NullComplexParent();
+            CustomCollection = new CustomCollection_NullCollectionsTestsModel<int> { 42 };
+            NullableColors = new Color?[] { Color.Red };
+        }
+
+        public int ID { get; set; }
+
+        public int[] Array { get; set; }
+
+        public IEnumerable<int> IEnumerable { get; set; }
+
+        public IList<int> IList { get; set; }
+
+        public ICollection<int> ICollection { get; set; }
+
+        public Collection<int> Collection { get; set; }
+
+        public List<int> List { get; set; }
+
+        public IEnumerable<NullComplexChild> ComplexCollection { get; set; }
+
+        public NullComplexParent ParentComplex { get; set; }
+
+        //public Vehicle[] Vehicles { get; set; }
+
+        //public Vehicle Vehicle { get; set; }
+
+        public CustomCollection_NullCollectionsTestsModel<int> CustomCollection { get; set; }
+
+        public IEnumerable<Color?> NullableColors { get; set; }
+    }
+
+    public class NullComplexChild
+    {
+        public int A { get; set; }
+    }
+
+    // we re-test all of the above properties on the entity (except ourself but including a different child complex 
+    // collection) because the serializer handles each parent type (including entity vs complex) through different 
+    // code paths
+    public class NullComplexParent
+    {
+        public NullComplexParent()
+        {
+            Array = new int[] { 42 };
+            IEnumerable = new int[] { 42 };
+            ICollection = new int[] { 42 };
+            IList = new int[] { 42 };
+            List = new List<int> { 42 };
+            Collection = new Collection<int> { 42 };
+            ComplexCollection = new NullComplexChild[] { new NullComplexChild { A = 42 } };
+            CustomCollection = new CustomCollection_NullCollectionsTestsModel<int> { 42 };
+            NullableColors = new Color?[] { Color.Red };
+        }
+
+        public int[] Array { get; set; }
+
+        public IEnumerable<int> IEnumerable { get; set; }
+
+        public IList<int> IList { get; set; }
+
+        public ICollection<int> ICollection { get; set; }
+
+        public Collection<int> Collection { get; set; }
+
+        public List<int> List { get; set; }
+
+        public IEnumerable<NullComplexChild> ComplexCollection { get; set; }
+
+        //public Vehicle[] Vehicles { get; set; }
```

Comment:
Your comment isn't clear.  Do you mean that NullCollectionsTestsModel and NullComplexParent look similar?  This is because I wanted to test all types of properties at both the entity and complex level since there are two different serializer paths.